### PR TITLE
New Feature - added All DBSupports to Knex

### PR DESCRIPTION
**Pull request by Jose-Perigolo**
- Update -new tests for Postgres and SQLite
- Added new qBuilder
- New package.json commands for testing 
- Knex needs SQLite and PG sdk in order to work, added in package.json
- Some tests still not passing using SQLite, making corrections.

### CODE REVIEW

File: dist/intern.d.ts [hunk 1]
```diff
@@ -1,27 +1,23 @@
 export declare class intern {
     static asyncMethod(f: any): (this: any, msg: any, done: any, meta: any) => any;
-    static findKnex(ent: any, q: any): Promise<any>;
-    static firstKnex(ent: any, q: any): Promise<any>;
-    static insertKnex(ent: any, data: any): Promise<any>;
-    static updateKnex(ent: any, data: any): Promise<any>;
-    static removeKnex(ent: any, q: any): Promise<any>;
-    static upsertKnex(ent: any, data: any, q: any): Promise<any>;
+    static findKnex(ent: any, q: any, knex: any): Promise<any>;
```
Comment (rjrodger) [inline dist/intern.d.ts]: let's make this api more "predictable" - see https://wiki.c2.com/?PrincipleOfLeastAstonishment

let's use the arg order: 
`knex, ent, q`
moving from left to right, from most general, to most specific

File: package.json [hunk 2]
```diff
@@ -45,20 +46,21 @@
     "knexjs"
   ],
   "peerDependencies": {
+    "knex": ">=2.4.2",
```
Comment (rjrodger) [inline package.json]: would `>=2` be sufficient?

File: package.json [hunk 3]
```diff
@@ -45,20 +46,21 @@
...
-    "knex": ">=2.4.2"
+    "seneca-entity": ">=18"
   },
   "devDependencies": {
-    "pg": ">=8.8.0",
     "@hapi/code": "9.0.1",
     "@hapi/lab": "25.0.1",
     "async": "3.2.4",
     "lab-transform-typescript": "3.0.1",
+    "pg": ">=8.8.0",
```
Comment (rjrodger) [inline package.json]: for more predicable builds, let's use fixed version numbers

File: src/intern.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import Q from './qbuilder'
+import qBuilder from './qbuilder'
```
Comment (rjrodger) [inline src/intern.ts]: `QBuilder`

File: src/intern.ts [hunk 5]
```diff
@@ -20,39 +20,93 @@ export class intern {
     }
   }
 
-  static async findKnex(ent: any, q: any): Promise<any> {
+  static async findKnex(ent: any, q: any, knex: any): Promise<any> {
```
Comment (rjrodger) [inline src/intern.ts]: `intern` should not be a class - it is never something we instantiate
just use a normal object with the knex functions are properties

File: src/intern.ts [hunk 6]
```diff
@@ -20,39 +20,93 @@ export class intern {
...
       data: intern.isObjectEmpty(filter) ? false : filter,
-      isArray: isQArray
+      isArray: isQArray,
+      sort,
+      skip,
+      ...(limit && {limit})
     }
     
-    const query = await Q.select(args)
+    const query = await qBuilder(knex).select(args)
```
Comment (rjrodger) [inline src/intern.ts]: intead of creating a new QBuilder each time, just create it once, after you get the knex object

File: src/knex-store.ts [hunk 7]
```diff
@@ -44,145 +35,92 @@ function knex_store(this: any, options: Options) {
...
-                newEnt.id = ent.id$
-              }
-              
-              const doCreate = await intern.insertKnex(newEnt, ctx)
-              
-              return doCreate
-              
-            } catch (err) {
-              return err
+    save: asyncMethod(async function (this: any, msg: any) {
```
Comment (rjrodger) [inline src/knex-store.ts]: remove asyncMethod - this has no real benefit in TS, and may break with future versions of Seneca - just set the function directly

**Review by rjrodger — CHANGES_REQUESTED** [review]

File: dist/intern.d.ts (hunk 1 above)
Comment (Jose-Perigolo) [reply dist/intern.d.ts]: Done. Wonderful, I learned a new principle : ) .

File: package.json (hunk 2 above)
Comment (Jose-Perigolo) [reply package.json]: Yep, I switched to 2.0.0

File: package.json (hunk 3 above)
Comment (Jose-Perigolo) [reply package.json]: Got it, changed to `"pg": "8.8.0"`

File: src/intern.ts (hunk 4 above)
Comment (Jose-Perigolo) [reply src/intern.ts]: PascalCase? Got it

File: src/intern.ts (hunk 6 above)
Comment (Jose-Perigolo) [reply src/intern.ts]: Hmmm, like this?
```
const builder = qBuilder(knex)

....

const query = await builder.select(args)
```

File: src/knex-store.ts (hunk 7 above)
Comment (Jose-Perigolo) [reply src/knex-store.ts]: Got it

File: src/intern.ts (hunk 5 above)
Comment (Jose-Perigolo) [reply src/intern.ts]: Ok! So much better that way!

**Pull request merged**